engine/src/main/java/io/camunda/zeebe/streamprocessor/records/RecordBatchEntry.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.streamprocessor.records;
+
+import static io.camunda.zeebe.engine.processing.streamprocessor.TypedEventRegistry.EVENT_REGISTRY;
+
+import io.camunda.zeebe.protocol.impl.record.RecordMetadata;
+import io.camunda.zeebe.protocol.impl.record.UnifiedRecordValue;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.Intent;
+import io.camunda.zeebe.util.ReflectUtil;
+import io.camunda.zeebe.util.buffer.BufferWriter;
+import org.agrona.DirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+
+public record RecordBatchEntry(
+    long key, int sourceIndex, RecordMetadata recordMetadata, DirectBuffer recordValueBuffer)
+    implements UnmodifiableRecordBatchEntry {
+
+  @Override
+  public UnifiedRecordValue recordValue() {
+    final UnifiedRecordValue unifiedRecordValue =
```

Comment:
:wrench: Shouldn't we do that in the constructor? We can replace the field `recordValueBuffer` with a `UnifiedRecordValue recordValue`, and already convert it in the static factory method below?

From the API, I would expect the `recordValue` to not create anything. Plus, it would probably help debuggability since the toString will show the toString of the record value, and not just the buffer.

WDYT?